### ISSUE TITLE
Gradle: Fail Build if not formatted (with compileJava.dependsOn('spotlessApply'))

### ISSUE THREAD

**Issue by vbrandl** [diffplug/spotless#453]
I am using Gradle 5.2 with version `3.24.2` of the spotless plugin.

I want to format JSF XHTML files using the eclipse formatter, using the following config:

```
...
if (!isCI) {
  project.afterEvaluate {
    compileJava.dependsOn('spotlessApply')
  }
}
...
spotless {
  format 'xml', {
    target '**/*.xhtml'
    eclipseWtp('html')
    indentWithSpaces(2)
    endWithNewline()
    trimTrailingWhitespace()
    paddedCell()
    encoding 'UTF-8'
  }
}
```

The goal is to automatically format the source on every build, when not in the CI environment but fail, if in the CI environment and the source is not formatted.

Formatting on every build works due to the `afterEvaluate` block, but when running `./gradlew check` or `./gradlew spotlessCheck` on unformatted sources, either in the CI environment or with the `afterEvaluete` block commented out, the build does not fail.

What am I missing? I want `check` and `spotlessCheck` to fail, if the sources are not formatted...

**Comment by nedtwigg** [comment]
In this case, I don't think we can help without a public repo to reproduce.

Generally speaking, when people have tried to do `something.dependsOn('spotlessApply')`, they are often surprised that `check` never fails on their machine, but then fails in CI.  And that's because they commit badly-formatted files, then run a build which passes (because spotlessApply changes the files after they were commited), and then they push, and it fails on CI.

It looks like you are carefully avoiding the pitfall above.  It looks to me like your code should work.  If you make a public repo that reproduces this issue, I'll take a look.  But my best guess is that there's some confusion, only because we have had so many issues in the past with this sort of confusion.

**Comment by nedtwigg** [comment]
Awesome!  I'm able to repro on my machine, thanks for the example.  Surprisingly, I think this turns out to be a dup (but much smaller and therefore easier to work with) of #371 

TLDR, this will be fixed if you turn off paddedCell.  But this should work, and this is a great testcase to help us fix the underlying bug.

**Comment by vbrandl** [comment]
Turning off `paddedCell` works in the minimal example but results in an error on the production repo. So this seems like a deadlock until the underlying issue is fixed?

**Comment by nedtwigg** [comment]
Yep.  Lots of formatters have non-idempotent corner-cases.  Looks like your prod repo has one of them.  That's what paddedCell is for, but clearly something about it is broken.  Your test repo is a perfect testcase for fixing it, and I'm happy to merge any PR's, but I won't get to it until this weekend.  I expect we should be able to ship a fix on Monday.

**Comment by nedtwigg** [comment]
Fixed in `3.24.3`.  Thanks for the testcase!
